### PR TITLE
rust => 1.66.0

### PR DESCRIPTION
**Pull request by satmandu**
Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=rust CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/rust.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def self.install
     ENV['RUSTUP_HOME'] = "#{CREW_DEST_PREFIX}/share/rustup"
     default_host = ARCH == 'aarch64' || ARCH == 'armv7l' ? 'armv7-unknown-linux-gnueabihf' : "#{ARCH}-unknown-linux-gnu"
     downloader 'https://sh.rustup.rs', '173f4881e2de99ba9ad1acb59e65be01b2a44979d83b6ec648d0d22f8654cbce', 'rustup.sh'
```
Comment (uberhacker) [inline packages/rust.rb:34]: Add to `source_url` and `git_hashtag` instead?

**Review by uberhacker — CHANGES_REQUESTED** [review]
The idea is to reduce complexity.

File: packages/rust.rb (hunk 1 above)
Comment (uberhacker) [reply packages/rust.rb:34]: Change `system "install -Dm644 ...` to `FileUtils.install ... mode: 0o644`?

Comment (satmandu) [reply packages/rust.rb:34]: Sure... changing....

File: packages/rust.rb [hunk 2]
```diff
@@ -31,13 +32,12 @@ def self.install
...
-    downloader 'https://sh.rustup.rs', '173f4881e2de99ba9ad1acb59e65be01b2a44979d83b6ec648d0d22f8654cbce', 'rustup.sh'
-    system "sed -i 's,\$(mktemp -d 2>/dev/null || ensure mktemp -d -t rustup),#{CREW_PREFIX}/tmp,' rustup.sh"
+    system "sed -i 's,$(mktemp -d 2>/dev/null || ensure mktemp -d -t rustup),#{CREW_PREFIX}/tmp,' rustup-init.sh"
     FileUtils.mkdir_p(CREW_DEST_HOME)
     FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/bin")
     FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/share/cargo")
     FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/share/rustup")
-    system "RUSTFLAGS='-Clto=thin' bash ./rustup.sh -y --no-modify-path --default-host #{default_host} --default-toolchain #{@_ver} --profile minimal"
+    system "RUSTFLAGS='-Clto=thin' bash ./rustup-init.sh -y --no-modify-path --default-host #{default_host} --default-toolchain #{@_ver} --profile minimal"
     system "install -Dm644 #{CREW_DEST_PREFIX}/share/rustup/toolchains/#{@_ver}-#{default_host}/etc/bash_completion.d/cargo #{CREW_DEST_PREFIX}/share/bash-completion/completions/cargo"
```
Comment (uberhacker) [inline packages/rust.rb:41]: Change to `FileUtils.install ...`?

**Review by uberhacker — CHANGES_REQUESTED** [review]

**Review by uberhacker — APPROVED** [review]

**Pull request merged**